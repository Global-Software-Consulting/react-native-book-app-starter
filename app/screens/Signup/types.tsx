export interface ISignupData {
    firstName: string;
    lastName: string;
    email: string;
    gender: string;
    password: string;
}
