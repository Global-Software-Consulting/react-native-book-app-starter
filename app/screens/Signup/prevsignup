<View>
<View style={styles.infoView}>
    <TextInput
        style={styles.inputField}
        textContentType="emailAddress"
        keyboardType="email-address"
        placeholder={t('Enter your email')}
        onChangeText={(text) => setEmail(text)}
    />
</View>
</View>
