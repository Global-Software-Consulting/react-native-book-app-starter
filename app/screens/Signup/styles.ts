import { StyleSheet } from 'react-native';
import { useTheme } from 'react-native-paper';
import { getPercentageHeight, getPercentageWidth } from 'utils/dimentionUtil';
export const useStyles = () => {
    const height = getPercentageHeight();
    const width = getPercentageWidth();
    const theme = useTheme();
    const styles = StyleSheet.create({
        container: {
            flex: 1,
            backgroundColor: 'white',
        },
        dropDown: {
            width: width('80%'),
            alignSelf: 'center',
            borderRadius: 20,
            marginTop: 10,
        },
        displayPicture: {
            height: height('15%'),
            width: width('30%'),
            marginVertical: 20,
            borderRadius: 40,
            backgroundColor: 'transparent',
        },
        horizontalRuler: {
            borderBottomWidth: 2,
            marginTop: 3,
            color: 'gray',
            borderBottomColor: 'red',
        },
        mainHeading: {
            fontWeight: 'bold',
            color: theme.colors.text,
            fontSize: 25,
            marginTop: 20,
            marginBottom: 20,
        },
        subHeading: {
            fontWeight: 'bold',
            color: theme.colors.text,
        },
        inputField: {
            width: width('50%'),
            color: 'black',
        },
        genderPicker: {
            width: width('60%'),
            alignSelf: 'center',
            borderRadius: 20,
            marginTop: 10,
        },
        infoView: {
            marginBottom: 10,
            borderColor: theme.colors.text,
            borderRadius: 20,
            alignItems: 'center',
            justifyContent: 'center',
            width: width('80%'),
            height: 50,
        },
        cardView: {
            backgroundColor: 'white',
            width: width('90%'),
            zIndex: 5,
            borderRadius: 20,
            marginTop: -30,
            alignSelf: 'center',
            justifyContent: 'center',
        },
        inputView: {
            marginBottom: 10,
            borderColor: theme.colors.text,
            borderRadius: 20,
            alignItems: 'center',
            justifyContent: 'center',
            width: width('80%'),
            height: height('5%'),
        },
        editView: {
            backgroundColor: '#00416A',
            alignContent: 'center',
            alignSelf: 'center',
            justifyContent: 'space-between',
            flexDirection: 'row',
            marginTop: 30,
            borderRadius: 20,
            marginBottom: 20,
        },
        editButton: {
            alignSelf: 'center',
            justifyContent: 'center',
            marginTop: 5,
        },
        linearGradient: {
            height: height('15%'),
            alignItems: 'center',
            justifyContent: 'center',
            width: width('100%'),
        },
        imgView: {
            backgroundColor: 'white',
            width: width('90%'),
            zIndex: 5,
            borderRadius: 20,
            marginTop: -30,
            alignSelf: 'center',
            justifyContent: 'center',
        },
        img: { alignSelf: 'center', position: 'relative' },
        signUpText: {
            fontSize: 30,
            fontWeight: '500',
            alignSelf: 'center',
            justifyContent: 'center',
            marginTop: 20,
            color: '#3A3B3C',
        },
        submit: {
            height: 40,
            width: width('60%'),
            marginTop: 1,
        },
        submitView: {
            backgroundColor: '#00416A',
            alignContent: 'center',
            alignSelf: 'center',
            justifyContent: 'space-between',
            flexDirection: 'row',
            marginTop: 30,
            borderRadius: 20,
            marginBottom: 20,
        },
        button: { height: 40, width: width('60%'), marginTop: 5 },
        activity: {
            marginLeft: width('50%'),
            alignSelf: 'center',
            position: 'absolute',
        },
    });
    return styles;
};
