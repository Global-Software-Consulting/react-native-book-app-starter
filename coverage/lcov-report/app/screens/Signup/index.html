
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for app/screens/Signup</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> app/screens/Signup</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>17/34</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">14.28% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/14</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>17/34</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="index.tsx"><a href="index.tsx.html">index.tsx</a></td>
	<td data-value="42.85" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 42%"></div><div class="cover-empty" style="width: 58%"></div></div>
	</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="28" class="abs low">12/28</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="7.69" class="pct low">7.69%</td>
	<td data-value="13" class="abs low">1/13</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="28" class="abs low">12/28</td>
	</tr>

<tr>
	<td class="file high" data-value="styles.ts"><a href="styles.ts.html">styles.ts</a></td>
	<td data-value="83.33" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 83%"></div><div class="cover-empty" style="width: 17%"></div></div>
	</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Nov 10 2021 14:36:08 GMT+0500 (Pakistan Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    