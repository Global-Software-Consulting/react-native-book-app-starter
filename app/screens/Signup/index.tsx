import images from 'config/images';
import { IAppState } from 'models/reducers/appReducers';
import React, { useEffect, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import { ActivityIndicator, Image, TextInput, View } from 'react-native';
import DropDownPicker from 'react-native-dropdown-picker';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import LinearGradient from 'react-native-linear-gradient';
import { Button, Text } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import { useStyles } from 'screens/Signup/styles';
import * as appActions from 'store/actions/appActions';
import * as loginActions from 'store/actions/loginActions';
import { ISignupData } from './types';
import Toast from 'react-native-simple-toast';

const Signup: React.FC = () => {
    const dispatch = useDispatch();
    //defining states
    const [open, setOpen] = useState<boolean>(false);
    const { t } = useTranslation();
    const isLoading = useSelector(
        (state: { loadingReducer: IAppState }) => state.loadingReducer.isLoading,
    );
    const gender = [
        { label: 'Male', value: 'male' },
        { label: 'Female', value: 'female' },
    ];
    const {
        control,
        handleSubmit,
        formState: { errors },
    } = useForm();

    const validate = (text: string) => {
        let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w\w+)+$/;
        return reg.test(text);
    };

    const performSignUp = async (data: ISignupData) => {
        if (validate(data.email)) {
            dispatch(loginActions.signupRequest(data));
        } else {
            Toast.show('Incorrect email format');
        }
    };

    const onSubmit = (data: ISignupData) => {
        performSignUp(data);
    };

    const styles = useStyles();

    return (
        <KeyboardAwareScrollView style={{ flex: 1 }}>
            <LinearGradient
                colors={['#00416A', '#00416A', '#E4E5E6']}
                start={{ x: 0.0, y: 0.5 }}
                end={{ x: 0.1, y: 3.0 }}
                locations={[0, 0.5, 0.6]}
                style={styles.linearGradient}
            />
            <View style={styles.cardView}>
                <Image source={images.app.logo} style={styles.img} />
                <Text style={styles.signUpText}>{t('Sign up')}</Text>

                <View style={styles.inputView}>
                    <Controller
                        control={control}
                        rules={{
                            required: true,
                        }}
                        render={({ field: { onChange, value } }) => (
                            <TextInput
                                placeholder={t('Enter your first name')}
                                autoCapitalize="none"
                                autoCorrect={false}
                                style={styles.inputField}
                                value={value}
                                onChangeText={(text) => onChange(text)}
                            />
                        )}
                        name="firstName"
                        defaultValue=""
                    />
                    {errors.firstName && (
                        <Text style={{ color: 'red' }}>{t('First name is required')}</Text>
                    )}
                </View>

                <View style={styles.inputView}>
                    <Controller
                        control={control}
                        rules={{
                            required: true,
                        }}
                        render={({ field: { onChange, value } }) => (
                            <TextInput
                                placeholder={t('Enter your last name')}
                                autoCapitalize="none"
                                autoCorrect={false}
                                style={styles.inputField}
                                value={value}
                                onChangeText={(text) => onChange(text)}
                            />
                        )}
                        name="lastName"
                        defaultValue=""
                    />
                    {errors.lastName && (
                        <Text style={{ color: 'red' }}>{t('Last name is required')}</Text>
                    )}
                </View>

                <View style={styles.inputView}>
                    <Controller
                        control={control}
                        rules={{
                            required: true,
                        }}
                        render={({ field: { onChange, value } }) => (
                            <TextInput
                                placeholder={t('Enter your email')}
                                autoCapitalize="none"
                                textContentType="emailAddress"
                                keyboardType="email-address"
                                autoCorrect={false}
                                style={styles.inputField}
                                value={value}
                                onChangeText={(text) => onChange(text)}
                            />
                        )}
                        name="email"
                        defaultValue=""
                    />
                    {errors.email && <Text style={{ color: 'red' }}>{t('Email is required')}</Text>}
                </View>

                <View style={styles.inputView}>
                    <Controller
                        control={control}
                        rules={{
                            required: true,
                        }}
                        render={({ field: { onChange, value } }) => (
                            <TextInput
                                placeholder={t('Enter password')}
                                secureTextEntry={true}
                                autoCapitalize="none"
                                autoCorrect={false}
                                style={styles.inputField}
                                value={value}
                                onChangeText={(text) => onChange(text)}
                            />
                        )}
                        name="password"
                        defaultValue=""
                    />
                    {errors.lastName && (
                        <Text style={{ color: 'red' }}>{t('Password is required')}</Text>
                    )}
                </View>

                <Controller
                    control={control}
                    rules={{
                        required: true,
                    }}
                    render={({ field: { onChange, value } }) => (
                        <DropDownPicker
                            placeholder={t('Please select gender')}
                            open={open}
                            value={value}
                            items={gender}
                            setOpen={setOpen}
                            setValue={onChange}
                            style={styles.genderPicker}
                            dropDownContainerStyle={styles.dropDown}
                        />
                    )}
                    name="gender"
                    defaultValue=""
                />
                {errors.value && <Text style={{ color: 'red' }}>{t('Please provide gender')}</Text>}

                <View style={styles.submitView}>
                    <Button
                        onPress={handleSubmit(onSubmit)}
                        disabled={isLoading}
                        style={styles.submit}>
                        <Text style={{ color: 'white' }}>{t('Sign up')}</Text>
                    </Button>
                    {isLoading && <ActivityIndicator color="white" style={styles.activity} />}
                </View>
            </View>
        </KeyboardAwareScrollView>
    );
};

export default Signup;
