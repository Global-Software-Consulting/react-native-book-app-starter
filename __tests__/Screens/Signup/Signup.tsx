import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import * as redux from 'react-redux'
import { act } from 'react-test-renderer'
import Signup from '../../../app/screens/Signup/index'
import { useDispatch } from 'react-redux'

jest.mock('@react-navigation/core', () => {
    return {
        __esModule: true,
        A: true,
        useNavigation: jest.fn(),
        navigation: { navigate: jest.fn() },

        default: 'mockedDefaultExport',
    }
})
jest.mock('./../../../app/config/images', () => {
    return {
        __esModule: true,
        A: true,
        images: {
            app: {
                logo: 'abc',
            },
        },
        default: 'mockedDefaultExport',
    }
})
jest.mock('react-native-keyboard-aware-scroll-view', () => {
    return {
        __esModule: true,
        A: true,
        KeyboardAwareScrollView: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('react-native-linear-gradient', () => {
    return {
        __esModule: true,
        A: true,
        LinearGradient: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('./../../../app/config/images', () => {
    return {
        __esModule: true,
        A: true,
        app: { logo: jest.fn() },
        default: 'mockedDefaultExport',
    }
})

describe('Component testing', () => {
    test('Snapshot with loading as true', async () => {
        const tree = render(<Signup />)
        act(() => {
            const state = { loadingReducer: { isLoading: true } }
            const isFocussed = jest.spyOn(redux, 'useSelector').mockImplementation((val) => {
                val(state)
                isFocussed.mockClear()
            })
        })
        expect(tree).toMatchSnapshot()
    })
    test('Snapshot with loading as true', async () => {
        const tree = render(<Signup />)
        act(() => {
            const state = { loadingReducer: { isLoading: false } }
            const isFocussed = jest.spyOn(redux, 'useSelector').mockImplementation((val) => {
                val(state)
                isFocussed.mockClear()
            })
        })
        expect(tree).toMatchSnapshot()
    })
    test('firstname text change', async () => {
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.changeText(getByTestId('firstname'))
        })
    })
    test('lastname text change', async () => {
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.changeText(getByTestId('lastname'))
        })
    })
    test('email text change', async () => {
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.changeText(getByTestId('email'))
        })
    })
    test('password text change', async () => {
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.changeText(getByTestId('password'))
        })
    })
    test('submit details', async () => {
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.press(getByTestId('submit'))
        })
    })
    test('submit with wrong email format', async () => {
        const dispatch = jest.fn()
        useDispatch.mockReturnValue(jest.fn())
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.changeText(getByTestId('firstname'), 'dsa')
            fireEvent.changeText(getByTestId('lastname'), 'fdsa')

            fireEvent.changeText(getByTestId('email'), 'abcd')
            fireEvent.changeText(getByTestId('password'), '123frfr')

            fireEvent.press(getByTestId('submit'))
        })
    })

    test('submit with proper information filled in', async () => {
        const { getByTestId } = render(<Signup />)
        act(() => {
            fireEvent.changeText(getByTestId('firstname'), 'dsa')
            fireEvent.changeText(getByTestId('lastname'), 'fdsa')

            fireEvent.changeText(getByTestId('email'), 'abc@gmail.com')
            fireEvent.changeText(getByTestId('password'), '123frfr')

            fireEvent.press(getByTestId('submit'))
        })
    })
})
