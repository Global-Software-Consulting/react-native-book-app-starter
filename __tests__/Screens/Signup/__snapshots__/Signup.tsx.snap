// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot with loading as true 1`] = `
<View>
  <mockedDefaultExport
    colors={
      Array [
        "#00416A",
        "#00416A",
        "#E4E5E6",
      ]
    }
    end={
      Object {
        "x": 0.1,
        "y": 3,
      }
    }
    locations={
      Array [
        0,
        0.5,
        0.6,
      ]
    }
    start={
      Object {
        "x": 0,
        "y": 0.5,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "center",
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 20,
        "justifyContent": "center",
        "marginTop": -30,
        "width": 675,
        "zIndex": 5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "alignSelf": "center",
            "color": "#3A3B3C",
            "fontSize": 30,
            "fontWeight": "500",
            "justifyContent": "center",
            "marginTop": 20,
          },
        ]
      }
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="firstname"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="lastname"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="email"
      textContentType="emailAddress"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="password"
      value=""
    />
    <View
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#00416A",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginTop": 30,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "height": 40,
            "marginTop": 1,
            "minWidth": 64,
            "width": 450,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
          testID="submit"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "white",
                    },
                  ]
                }
              />
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component testing Snapshot with loading as true 2`] = `
<View>
  <mockedDefaultExport
    colors={
      Array [
        "#00416A",
        "#00416A",
        "#E4E5E6",
      ]
    }
    end={
      Object {
        "x": 0.1,
        "y": 3,
      }
    }
    locations={
      Array [
        0,
        0.5,
        0.6,
      ]
    }
    start={
      Object {
        "x": 0,
        "y": 0.5,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "center",
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 20,
        "justifyContent": "center",
        "marginTop": -30,
        "width": 675,
        "zIndex": 5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "alignSelf": "center",
            "color": "#3A3B3C",
            "fontSize": 30,
            "fontWeight": "500",
            "justifyContent": "center",
            "marginTop": 20,
          },
        ]
      }
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="firstname"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="lastname"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="email"
      textContentType="emailAddress"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "margin": 10,
          "width": 450,
        }
      }
      testID="password"
      value=""
    />
    <View
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#00416A",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginTop": 30,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "height": 40,
            "marginTop": 1,
            "minWidth": 64,
            "width": 450,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
          testID="submit"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "white",
                    },
                  ]
                }
              />
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
